x-pack/plugins/observability_solution/slo/public/components/slo/burn_rate/alert_details/components/custom_panels/custom_kql/log_rate_analysis_panel.tsx

```diff
@@ -0,0 +1,285 @@
+import { useFetchDataViews } from '@kbn/observability-plugin/public';
+import { colorTransformer, Color } from '@kbn/observability-shared-plugin/common';
+import { KQLCustomIndicator } from '@kbn/slo-schema';
+import { i18n } from '@kbn/i18n';
+import type { Message } from '@kbn/observability-ai-assistant-plugin/public';
+import { BurnRateAlert, BurnRateRule } from '../../../alert_details_app_section';
+import { useKibana } from '../../../../../../../utils/kibana_react';
+import { getESQueryForLogRateAnalysis } from './helpers/log_rate_analysis_query';
+
+interface Props {
+  slo: GetSLOResponse;
+  alert: BurnRateAlert;
+  rule: BurnRateRule;
+}
+
+interface SignificantFieldValue {
+  field: string;
+  value: string | number;
+  docCount: number;
+  pValue: number | null;
+}
+
+export function LogRateAnalysisPanel({ slo, alert, rule }: Props) {
+  const services = useKibana().services;
+  const { dataViews: dataViewsService, observabilityAIAssistant } = services;
+  const ObservabilityAIAssistantContextualInsight =
+    observabilityAIAssistant?.ObservabilityAIAssistantContextualInsight;
+  const [dataView, setDataView] = useState<DataView | undefined>();
+  const [esSearchQuery, setEsSearchQuery] = useState<QueryDslQueryContainer | undefined>();
+  const [logRateAnalysisParams, setLogRateAnalysisParams] = useState<
+    | { logRateAnalysisType: LogRateAnalysisType; significantFieldValues: SignificantFieldValue[] }
+    | undefined
+  >();
+  const params = slo.indicator.params as KQLCustomIndicator['params'];
+  const groupBy = slo.groupBy;
+  const groupings = slo.groupings;
+  const { index } = params;
+  const { data: dataViews = [] } = useFetchDataViews();
+
+  useEffect(() => {
+    const getDataView = async () => {
+      const getDataViewByIndexPattern = (indexPattern: string) =>
+        dataViews.find((dataView0) => dataView0.title === indexPattern);
+
+      const dataViewId = getDataViewByIndexPattern(index)?.id;
+      if (dataViewId) {
+        const sloDataView = await dataViewsService.get(dataViewId);
+        setDataView(sloDataView);
+        getQuery();
+      }
+    };
+
+    const getQuery = () => {
+      const esSearchRequest = getESQueryForLogRateAnalysis(
+        params,
+        groupBy,
+        groupings
+      ) as QueryDslQueryContainer;
+      if (esSearchRequest) {
+        setEsSearchQuery(esSearchRequest);
+      }
+    };
+    getDataView();
+  }, [index, dataViews, params, dataViewsService, groupBy, groupings]);
+
+  // Identify `intervalFactor` to adjust time ranges based on alert settings.
+  // The default time ranges for `initialAnalysisStart` are suitable for a `1m` lookback.
+  // If an alert would have a `5m` lookback, this would result in a factor of `5`.
+  const lookbackDuration =
+    alert.fields['kibana.alert.rule.parameters'] &&
+    alert.fields['kibana.alert.rule.parameters'].timeSize &&
+    alert.fields['kibana.alert.rule.parameters'].timeUnit
```

Comment:
We don't have these parameters in the burn rate alert as data.

Reply:
Yep I need to remove this!